tests/app/main/test_frameworks.py

```diff
@@ -5758,3 +5757,209 @@ def test_false_values(self, falsey_value):
+        self.data_api_client_patch.stop()
+        super().teardown_method(method)
+
+    @pytest.mark.parametrize(
+        ('framework_slug', 'on_framework', 'status_code'),
+        (
+            ('g-cloud-11', True, 404),
+            ('g-cloud-12', True, 200),
+            ('g-cloud-12', False, 400),
+        )
+    )
+    def test_only_works_for_supported_frameworks(self, framework_slug, on_framework, status_code):
+        self.login()
+        self.data_api_client.get_framework.return_value = self.framework(status='standstill',
+                                                                         slug=framework_slug,
+                                                                         framework_agreement_version="1")
+        self.data_api_client.find_draft_services_by_framework.return_value = {
+            'meta': {'total': 1}
+        }
+        self.data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+            on_framework=on_framework)
+
+        res = self.client.get(f"/suppliers/frameworks/{framework_slug}/sign-framework-agreement")
+        assert res.status_code == status_code
+
+    def test_shows_error_messages(self):
+        self.login()
+        self.data_api_client.get_framework.return_value = self.framework(status='standstill',
+                                                                         slug='g-cloud-12',
+                                                                         framework_agreement_version="1")
+        self.data_api_client.find_draft_services_by_framework.return_value = {
+            'meta': {'total': 1}
+        }
+        self.data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+            on_framework=True)
+
+        res = self.client.post("/suppliers/frameworks/g-cloud-12/sign-framework-agreement", data={})
+        assert res.status_code == 400
+        text = res.get_data(as_text=True)
+        assert 'Enter your full name.' in text
+        assert 'Enter your role in the company.' in text
+        assert 'You must accept the terms and conditions of the Framework Agreement.' in text
+
+    def test_post_signs_agreement(self):
+        self.data_api_client.create_framework_agreement.return_value = {"agreement": {"id": 789}}
+        self.data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+            on_framework=True)
+        self.data_api_client.find_draft_services_by_framework.return_value = {
+            'meta': {'total': 1}
+        }
+        self.data_api_client.get_framework.return_value = self.framework(status='standstill',
+                                                                         slug='g-cloud-12',
+                                                                         framework_agreement_version="1")
+
+        self.login()
+        res = self.client.get("/suppliers/frameworks/g-cloud-12/sign-framework-agreement")
```

Comment:
Do we need the GET request here?

Reply:
see https://github.com/alphagov/digitalmarketplace-supplier-frontend/pull/1239#discussion_r476572666